zerver/lib/test_classes.py

```diff
@@ -1611,6 +1611,41 @@ def subscribe_via_post(
             self.assert_json_success(result)
         return result
 
+    # Create a stream by making an API request
+    def create_channel_via_post(
+        self,
+        user: UserProfile,
+        subscribers: list[str] | list[int] | None = None,
+        channel: dict[str, str] | None = None,
+        extra_post_data: Mapping[str, Any] = {},
+        invite_only: bool = False,
+        is_web_public: bool = False,
+        **extra: str,
+    ) -> "TestHttpResponse":
+        if channel is None:
+            channel = {"name": "new_channel"}
+        if subscribers is None:
+            subscribers = [user.id]
+
+        post_data = {
+            "channel": orjson.dumps(channel).decode(),
+            "subscribers": orjson.dumps(subscribers).decode(),
+            "is_web_public": orjson.dumps(is_web_public).decode(),
+            "invite_only": orjson.dumps(invite_only).decode(),
+        }
+
+        post_data.update(extra_post_data)
+        with self.artificial_transaction_savepoint():
```

Comment:
Are we certain this function actually needs `artificial_transaction_savepoint`? I was unsure if that was just required for the existing code path.